recipes/librdkafka/all/conanfile.py

```diff
@@ -117,8 +114,18 @@ def package_info(self):
         self.cpp_info.components["rdkafka"].names["cmake_find_package_multi"] = "rdkafka"
         self.cpp_info.components["rdkafka"].names["pkg_config"] = "rdkafka"
         self.cpp_info.components["rdkafka"].libs = ["rdkafka"]
-        if self.settings.compiler == "Visual Studio":
-            self.cpp_info.components["rdkafka"].system_libs.extend(["crypt32", "ws2_32"])
+        if self.options.zlib:
+            self.cpp_info.components["rdkafka"].requires.append("zlib::zlib")
+        if self.options.zstd:
+            self.cpp_info.components["rdkafka"].requires.append("zstd::zstd")
+        if self.options.ssl:
+            self.cpp_info.components["rdkafka"].requires.append("openssl::openssl")
+        if self.options.sasl and self.settings.os != "Windows":
+            self.cpp_info.components["rdkafka"].requires.append("cyrus-sasl::cyrus-sasl")
+        if self.options.lz4:
+            self.cpp_info.components["rdkafka"].requires.append("lz4::lz4")
+        if self.settings.os == "Windows":
+            self.cpp_info.components["rdkafka"].system_libs.extend(["crypt32", "ws2_32", "secur32"])
```

Comment:
Finer grained (without breaking anything I hope):

```suggestion
            self.cpp_info.components["rdkafka"].system_libs = ["ws2_32", "secur32"]
            if self.options.ssl:
                self.cpp_info.components["rdkafka"].system_libs.append("crypt32")
```

Looking at source code and build files:
- `ws2_32` is required in `rdkafka_transport.c` (always included in the lib): https://github.com/edenhill/librdkafka/blob/515ed19ab6dea5b459f36575f430cfefba043a90/src/rdkafka_transport.c#L29
- `secur32` is required in `rdkafka_sasl_win32.c`: https://github.com/edenhill/librdkafka/blob/5f1fc42b2eed4433e217bea9155c84d3b1d83952/src/rdkafka_sasl_win32.c#L46
  `rdkafka_sasl_win32.c` is always compiled on Windows: https://github.com/edenhill/librdkafka/blob/8fc678518738c6b80380326dad86ef62228f87f0/src/CMakeLists.txt#L83
- `crypt32` is required in `rdkafka_ssl.c`: https://github.com/edenhill/librdkafka/blob/cedb2917396ddb02a853330adf5030506abdebf3/src/rdkafka_ssl.c#L40
  `rdkafka_ssl.c` is compiled if `WITH_SSL ON`: https://github.com/edenhill/librdkafka/blob/8fc678518738c6b80380326dad86ef62228f87f0/src/CMakeLists.txt#L67